### PR TITLE
Autodoc

### PR DESCRIPTION
**Pull request by asanin-epfl**

### CODE REVIEW

File: ci/check_generated_docstrings.sh [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+if [[ -z $LIBCLANG_PATH ]]; then
+    echo "Expect a \$LIBCLANG_PATH in the environment, it should have the path to the libclang.so"
+    exit -1
+fi
+
+#VERSION=14c7d7002c71aba0623bee07f3258c2453f06262
+#PACKAGE=git+git://github.com/pybind/pybind11_mkdoc.git@$VERSION
+
+VERSION=0938515e0d4f9a5c2c5949dc46462175f54d6ea4
```
Comment (asanin-epfl) [inline ci/check_generated_docstrings.sh:27]: @mgeplf I kept here your fork. Should I better revert to the original one?

Comment (mgeplf) [reply ci/check_generated_docstrings.sh:27]: My fork is stale, I assume, best to point at the original one, I think.  They merged my change.

File: .github/workflows/docstring_check.yaml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+on:
+    push:
+
+jobs:
+  build:
+    name: docstring-check
+    runs-on: ubuntu-18.04
+    steps:
+        - name: Fetch repository
+          uses: actions/checkout@v2.7.1
```
Comment (asanin-epfl) [inline .github/workflows/docstring_check.yaml:11]: What version use here? I thought the latest MorphIO `v2.7.1` but seems this version is about something else?

File: ci/check_generated_docstrings.sh (hunk 1 above)
Comment (asanin-epfl) [reply ci/check_generated_docstrings.sh]: Thank you, I changed to `master`

File: .github/workflows/docstring_check.yaml (hunk 2 above)
Comment (tomdele) [reply .github/workflows/docstring_check.yaml:11]: This is the version of the `checkout action`:
https://github.com/actions/checkout

So nothing related to morphio.

Comment (asanin-epfl) [reply .github/workflows/docstring_check.yaml:11]: Thank you so much! Lazy me

File: ci/check_generated_docstrings.sh [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    $VENV/bin/pip install 'clang==9'  # keep in sync w/ .github/workflows/docstring_check.yaml
+    $VENV/bin/python -m pip install $PACKAGE
+fi
+
+DOCSTRING_PATH=./binds/python/generated/docstrings.h
+rm -f $DOCSTRING_PATH
+
+$VENV/bin/python -m pybind11_mkdoc \
+  -o $DOCSTRING_PATH \
+  ./include/morphio/mut/*.h \
```
Comment (asanin-epfl) [inline ci/check_generated_docstrings.sh:39]: `./include/morphio/**/*.h` didn't work, so I had to split on 3 separate lines.

**Comment by asanin-epfl** [comment]
> Why did you choose the 2.1.0 ?

I'm just following `libsonata` as an example.

**Comment by tomdele** [comment]
> I'm just following libsonata as an example.

then why did I use checkout@v2.1.0 ??? :D

Joke aside I think it makes sense to use the v2 which should the LTS if I understood correctly.

**Comment by asanin-epfl** [comment]
> then why did I use checkout@v2.1.0 ??? :D

No idea :D
I changed it to `v2`

**Comment by asanin-epfl** [comment]
Hooray, I made this PR! Thanks to Andrew, it was possible.

File: ci/check_generated_docstrings.sh [hunk 4]
```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/env bash
+
+# Check the generated docstrings
+#
+# pybind11 ships with mkdoc.py which extracts python docstrings
+# from the C++ doxygen comments. This is beneficial for maintaining
+# consistency and reducing duplication.
+#
+# In order to enable libsonata to be built without enforcing clang
```
Comment (tomdele) [inline ci/check_generated_docstrings.sh:9]: typo here : `enable libsonata`

**Review by tomdele — APPROVED** [review]
I must say this is difficult to find out if something is not good with doxygen PR... 

So if you checked the output results and this is good then I am fine with the review.

**Comment by asanin-epfl** [comment]
retest this please

**Comment by asanin-epfl** [comment]
Thank you!

**Pull request merged**